lexers/embedded/robot.xml

```diff
@@ -0,0 +1,81 @@
+<lexer>
+  <config>
+    <name>Robot Framework</name>
+    <alias>robot</alias>
+    <filename>*.robot</filename>
+    <mime_type>text/x-robotframework</mime_type>
+  </config>
+  <rules>
+    <state name="root">
+      <rule>
+        <include state="whitespace"/>
+      </rule>
+      <rule pattern="^\*{3}\s*Settings\s*\*{3}">
+        <token type="Keyword"/>
+      </rule>
+      <rule pattern="^\*{3}\s*Test Cases\s*\*{3}">
+        <token type="Keyword"/>
+      </rule>
+      <rule pattern="^\*{3}\s*Variables\s*\*{3}">
+        <token type="Keyword"/>
+      </rule>
+      <rule pattern="^\*{3}\s*Keywords\s*\*{3}">
+        <token type="Keyword"/>
+      </rule>
+      <rule pattern="\[Documentation\]|\[Tags\]|\[Setup\]|\[Teardown\]|\[Template\]|\[Timeout\]">
+        <token type="NameTag"/>
+      </rule>
+      <rule pattern="#.*$">
+        <token type="Comment"/>
+      </rule>
+      <rule pattern="\$\{[^\}]+\}|\@{[^\}]+\}|\&\{[^\}]+\}">
+        <token type="NameVariable"/>
+      </rule>
+      <rule pattern="^\s+\[.*\]">
+        <token type="LiteralString"/>
+      </rule>
+      <rule pattern="\|\s*">
+        <token type="Punctuation"/>
+      </rule>
+      <rule>
+        <include state="keyword"/>
+      </rule>
+      <rule pattern=".">
+        <token type="Text"/>
+      </rule>
+    </state>
+    
+    <state name="keyword">
+      <rule pattern="^\s+\w+(\s+)+">
+        <token type="NameFunction"/>
+      </rule>
+      <rule pattern="\$\{[^\}]+\}|\@{[^\}]+\}|\&\{[^\}]+\}">
+        <token type="NameVariable"/>
+      </rule>
+      <rule pattern="(True|False|None|null|on|off)\b">
```

Comment:
One thing I noticed here is that this pattern will match portions of words. For example, when I tried it with a file that had the word `Documentation` in it, it highlighted the `on` at the end differently from the rest of the word.

Reply:
Good catch, it wrapped it in \b in order to match the whole word and not only parts of it.